ConsoleApp1/Program.cs

```diff
@@ -1,9 +1,478 @@
+            {
+                Console.WriteLine(ex.StackTrace);
+                throw;
+            }
+
+            Console.WriteLine("\nSearch result:");
+            DisplayPhoneBook(searchArray[0]);
+
+            for (int i = 1; i < searchArray.Length; i++)
+            {
+                if (searchArray[i].lastName.Contains(paramsSearch, StringComparison.OrdinalIgnoreCase) || searchArray[i].firstName.Contains(paramsSearch, StringComparison.OrdinalIgnoreCase) || searchArray[i].number.Contains(paramsSearch, StringComparison.OrdinalIgnoreCase) || searchArray[i].phoneNuber.Contains(paramsSearch, StringComparison.OrdinalIgnoreCase))
+                    DisplayPhoneBook(searchArray[i]);
+            }
+        }
+        static void Search(string paramsSearch)
+        {
+            (string number, string lastName, string firstName, string phoneNuber)[] searchArray = GetStringArrayFromFile(pathPhoneBook);
+
+            DisplayPhoneBook(searchArray[0]);
+
+            for (int i = 1; i < searchArray.Length; i++)
+            {
+                if (searchArray[i].lastName.Contains(paramsSearch, StringComparison.OrdinalIgnoreCase) || searchArray[i].firstName.Contains(paramsSearch, StringComparison.OrdinalIgnoreCase) || searchArray[i].number.Contains(paramsSearch, StringComparison.OrdinalIgnoreCase) || searchArray[i].phoneNuber.Contains(paramsSearch, StringComparison.OrdinalIgnoreCase))
+                    DisplayPhoneBook(searchArray[i]);
+            }
+        }
+        //
+        static void EditContact()
+        {
+            (string number, string lastName, string firstName, string phoneNuber)[] searchArray = GetStringArrayFromFile(pathPhoneBook);
+            string[] editContact = new string[searchArray.Length];
+            string paramsSearch;
+
+            Console.WriteLine("Enter (first name or last name or number) of the contact you want to change");
+            paramsSearch = Console.ReadLine() ?? throw new Exception();
+            Search(paramsSearch);
+
+            Console.Write("Select the order in Result:\t");
+            string indexOrder = Console.ReadLine() ?? throw new Exception();
+
+            Console.WriteLine(searchArray[int.Parse(indexOrder)]);
+
+            Console.WriteLine("What do you want to change:\n\t1. LastName\n\t2. First Name\n\t3. Phone Number");
+            ConsoleKeyInfo choice = Console.ReadKey();
+
+            switch (choice.Key)
+            {
+                case ConsoleKey.D1:
+                    Console.Clear();
+                    Console.WriteLine("Rewrite Last Name:\t");
+
+                    try
+                    {
+                        searchArray[int.Parse(indexOrder)].lastName = Console.ReadLine();
+                    }
+                    catch (Exception)
+                    {
+                        throw;
+                    }
+
+                    editContact = TransformArray(searchArray);
+
+                    try
```

Comment:
You have such duplicates through entire program, move write to the file logic to separated method